### PR TITLE
Add an auto-generated cast table to the docs.

### PR DESCRIPTION
**Pull request by vpetrovykh**
The table shows all the possible casts for the built-in scalars.

### CODE REVIEW

**Comment by 1st1** [comment]
Victor, can you add a screenshot of how this will look like in the docs?

**Comment by vpetrovykh** [comment]
I'll need to probably apply some styling to this to make it look sane, because the table is large. Vertical text for the top row should help keep things a bit nicer.

**Comment by 1st1** [comment]
> I'll need to probably apply some styling to this to make it look sane, because the table is large. Vertical text for the top row should help keep things a bit nicer.

That will be complicated. I suggest to generate a rst file with simple text/bullet list, that you can include (there should be a directive for that and if it doesn't work we can make it work)

**Comment by vpetrovykh** [comment]
I have uploaded a screenshot with a table and vertical text for top row. The corresponding changes to the edgedb-ui are very small (https://github.com/edgedb/ui/pull/316).

For an overall view of casts a table is better than many lists I think. Although maybe we can add a cast list at the end of each individual scalar page.

Screenshot_20210616_120419

**Comment by 1st1** [comment]
How does it look on mobile?

**Comment by 1st1** [comment]
Also It's not clear what 'impl' means.

**Comment by 1st1** [comment]
> "impl" - in addition to explicit and assignment cast, will be cast
  automatically when used in a context requiring the target type

Is a bit hard to parse, please add an example with `e.g. in ...`

I'd drop the legend under the table and just have a bullet list explaining what's what right above it.

**Comment by vpetrovykh** [comment]
> How does it look on mobile?

This is what iPhone 6/7/8 looks like in chrome:
Screenshot_20210616_125932
iPhone X:
Screenshot_20210616_130100
iPad:
Screenshot_20210616_130241

**Comment by vpetrovykh** [comment]
I've added more explicit descriptions of types of casts and provided examples for them.

**Comment by vpetrovykh** [comment]
The cast table has been trimmed to play nice with smaller screens (see https://github.com/edgedb/ui/pull/316).

**Review by 1st1 — APPROVED** [review]

**Comment by 1st1** [comment]
Thanks Victor!
